src/RedirectRequest.jl

```diff
@@ -21,6 +21,11 @@ function request(::Type{RedirectLayer{Next}},
     count = 0
     while true
 
+        # Verify the url before making the request. Verification is done in the redirect
+        # loop to also catch bad redirect URLs.
+        # TODO: At first thought that seemed like a good idea, but perhaps it is not?
+        verify_url(url)
```

Comment:
Thoughts on this? Missing host will break DNS lookup so that should be safe, and I suppose it does not hurt to verify the scheme either.